diff -ur reiser4progs-1.0.7.orig/libmisc/ui.c reiser4progs-1.0.7/libmisc/ui.c
--- reiser4progs-1.0.7.orig/libmisc/ui.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/libmisc/ui.c	2009-02-15 17:31:09.000000000 +0100
@@ -116,7 +116,7 @@
 				line_width = 0;
 			}
 
-			fprintf(stream, word);
+			fprintf(stream, "%s", word);
 
 			line_width += aal_strlen(word);
 
diff -ur reiser4progs-1.0.7.orig/librepair/add_missing.c reiser4progs-1.0.7/librepair/add_missing.c
--- reiser4progs-1.0.7.orig/librepair/add_missing.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/librepair/add_missing.c	2009-02-15 18:20:14.000000000 +0100
@@ -65,7 +65,7 @@
 	time_str = ctime(&am->stat.time);
 	time_str[aal_strlen(time_str) - 1] = '\0';
 	aal_stream_format(&stream, time_str);
-	aal_mess(stream.entity);
+	aal_mess("%s", (char *) stream.entity);
 	aal_stream_fini(&stream);
 }
 
diff -ur reiser4progs-1.0.7.orig/librepair/cleanup.c reiser4progs-1.0.7/librepair/cleanup.c
--- reiser4progs-1.0.7.orig/librepair/cleanup.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/librepair/cleanup.c	2009-02-15 18:53:37.000000000 +0100
@@ -26,7 +26,7 @@
 	time_str = ctime(&cleanup->stat.time);
 	time_str[aal_strlen(time_str) - 1] = '\0';
 	aal_stream_format(&stream, time_str);
-	aal_mess(stream.entity);
+	aal_mess("%s", (char *) stream.entity);
 	aal_stream_fini(&stream);
 }
 
diff -ur reiser4progs-1.0.7.orig/librepair/disk_scan.c reiser4progs-1.0.7/librepair/disk_scan.c
--- reiser4progs-1.0.7.orig/librepair/disk_scan.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/librepair/disk_scan.c	2009-02-15 18:14:53.000000000 +0100
@@ -45,7 +45,7 @@
 	time_str = ctime(&ds->stat.time);
 	time_str[aal_strlen(time_str) - 1] = '\0';
 	aal_stream_format(&stream, time_str);
-	aal_mess(stream.entity);
+	aal_mess("%s", (char *) stream.entity);
 	aal_stream_fini(&stream);
 }
 
diff -ur reiser4progs-1.0.7.orig/librepair/filter.c reiser4progs-1.0.7/librepair/filter.c
--- reiser4progs-1.0.7.orig/librepair/filter.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/librepair/filter.c	2009-02-15 17:32:41.000000000 +0100
@@ -591,7 +591,7 @@
 	time_str = ctime(&fd->stat.time);
 	time_str[aal_strlen(time_str) - 1] = '\0';
 	aal_stream_format(&stream, time_str);
-	aal_mess(stream.entity);
+	aal_mess("%s", (char *) stream.entity);
 	aal_stream_fini(&stream);
 }
 
diff -ur reiser4progs-1.0.7.orig/librepair/semantic.c reiser4progs-1.0.7/librepair/semantic.c
--- reiser4progs-1.0.7.orig/librepair/semantic.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/librepair/semantic.c	2009-02-15 18:26:50.000000000 +0100
@@ -791,7 +791,7 @@
 	time_str = ctime(&sem->stat.time);
 	time_str[aal_strlen(time_str) - 1] = '\0';
 	aal_stream_format(&stream, time_str);
-	aal_mess(stream.entity);
+	aal_mess("%s", (char *) stream.entity);
 	aal_stream_fini(&stream);
 }
 
diff -ur reiser4progs-1.0.7.orig/librepair/twig_scan.c reiser4progs-1.0.7/librepair/twig_scan.c
--- reiser4progs-1.0.7.orig/librepair/twig_scan.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/librepair/twig_scan.c	2009-02-15 18:21:39.000000000 +0100
@@ -119,7 +119,7 @@
 	time_str[aal_strlen(time_str) - 1] = '\0';
 	
 	aal_stream_format(&stream, time_str);
-	aal_mess(stream.entity);
+	aal_mess("%s", (char *) stream.entity);
 	aal_stream_fini(&stream);
 }
 
diff -ur reiser4progs-1.0.7.orig/progs/debugfs/browse.c reiser4progs-1.0.7/progs/debugfs/browse.c
--- reiser4progs-1.0.7.orig/progs/debugfs/browse.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/progs/debugfs/browse.c	2009-02-15 19:02:02.000000000 +0100
@@ -27,7 +27,7 @@
 		if (read <= 0)
 			break;
 
-		printf(buff);
+		printf("%s", buff);
 	}
 
 	return 0;
diff -ur reiser4progs-1.0.7.orig/progs/debugfs/print.c reiser4progs-1.0.7/progs/debugfs/print.c
--- reiser4progs-1.0.7.orig/progs/debugfs/print.c	2009-01-08 22:34:27.000000000 +0100
+++ reiser4progs-1.0.7/progs/debugfs/print.c	2009-02-15 18:57:18.000000000 +0100
@@ -24,7 +24,7 @@
 		if ((size = aal_stream_read(stream, buff, size)) <= 0)
 			return size;
 		
-		printf(buff);
+		printf("%s", buff);
 	}
 
 	return 0;
